inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+
+## Helper Functions
+
+Several helper functions have been added to EpiModel that make it clear the ```$``` indexing operations that are being done on the master ```dat``` object during simulation. These functions belong to the ```get``` and ```set``` family of functions and act on the various lists contained within the master ```dat``` object, including but not limited to the attribute, epi, control and param lists. These functions take as input, at mininum, the master ```dat``` object and the item (as a character string) to "get" or "set" ("age" or "group" in the attribute list or "inf.prob" in the parameter list for exampe).
+
+### Set Functions
+
+The ```set``` group of functions sets the value of an item on a sublist of the ```dat``` object; these functions are often used at the end of a modules sequence to update attributes or for tracking of epidemic statistics. As an example: 
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from one-Group infection module:
+dat <- set_epi(dat, "si.flow", at, nInf)
+```
+
+Here, we want to record the number of new infected - ```nInf``` -  at the current time step ```at```. We do so using the ```set_epi``` helper function and record the flow of succeptible to infected on to the master ```dat``` object. All set functions follow this naming scheme, with "set_" followed by the sublist of interest. Note: the set functions returns an updated ```dat``` object.
+
+### Get Functions
+
+The ```get``` group of functions retrieve the value of an item on a sublist of the ```dat``` objects, and are most often used when retrieving input parameters of ```init```, ```param``` and ```control``` objects or for querying current attribute or epidemic tracker lists. As an example:
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from One-Group arrivals module
+a.rate <- get_param(dat, "a.rate")
+nOld <- get_epi(dat, "num", index)
+```
+
+Here we are pulling the arrival rate, ```a.rate``` from the parameter list and querying the epidemic tracker list for the population size at the previous step. Note: the get functions will return a value or a vector of values as oppossed to the ```dat``` object returned by the set functions.
+
+Details of all available helper functions can be found using
+
+```{r echo = TRUE, eval = FALSE}
+?EpiModel::dat_get_set
+```
+
```

Comment:
missing changes about the `keep` control settings, how those with condensed into tergmLite argument

Reply:
Added section for tergmLite and folding in of `keep` into tergmLite.